Automattic/vip-go-mu-plugins
# Added site details data generation functionality

## Description

Add a filter to generate the site details index. 

Hook it to add the environment and core information.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Apply PR.
2. `lando test tests/config/test-sitedetailsindex.php `

-------------

1. Apply PR.
2. Add ` require_once( 'config/class-site-details-index.php' );` to the end of `000-vip-init.php`.
3. `lando wp shell`
4. Execute `Automattic\VIP\Config\Site_Details_Index::instance()->set_env_and_core( array() );`
5. The output should match what you'd expect for your environment.

------------

1. Apply PR.
2. `lando wp plugin install bbpress --activate`
3.  Add ` require_once( 'config/class-site-details-index.php' );` to the end of `000-vip-init.php`.
4. Execute `$site_details = Automattic\VIP\Config\Site_Details_Index::instance()->get_plugin_info();`
5. Verify that bbpress has `enabled_via_code` set to false.
6. Add `wpcom_vip_load_plugin( 'bbpress' );` to the end of `000-vip-init.php`
7. Execute `$site_details = Automattic\VIP\Config\Site_Details_Index::instance()->get_plugin_info();`
8. Verify that bbpress has `enabled_via_code` set to true.
